### ISSUE TITLE
Failed to parse expression with property access of "e"'s fields

### ISSUE THREAD

**Issue by OpportunityLiu** [josdejong/mathjs#2253]
When `e` is in an `x.e.y` pattern, a SyntaxError is raised.

![image](https://user-images.githubusercontent.com/13471233/122319011-93de0100-cf52-11eb-9901-b4db14797f76.png)

```
    var error = new SyntaxError(message + ' (char ' + c + ')');
                ^
SyntaxError: Digit expected, got "." (char 3)
    at createSyntaxError (***\node_modules\mathjs\lib\cjs\expression\parse.js:1748:17)
    at getToken (***\node_modules\mathjs\lib\cjs\expression\parse.js:408:17)
    at parseSymbol (***\node_modules\mathjs\lib\cjs\expression\parse.js:1296:7)
    at parseCustomNodes (***\node_modules\mathjs\lib\cjs\expression\parse.js:1282:12)
    at parseLeftHandOperators (***\node_modules\mathjs\lib\cjs\expression\parse.js:1200:12)
    at parsePow (***\node_modules\mathjs\lib\cjs\expression\parse.js:1178:12)
    at parseUnary (***\node_modules\mathjs\lib\cjs\expression\parse.js:1166:12)
    at parseRule2 (***\node_modules\mathjs\lib\cjs\expression\parse.js:1101:16)
    at parseImplicitMultiplication (***\node_modules\mathjs\lib\cjs\expression\parse.js:1070:12)
    at parseMultiplyDivide (***\node_modules\mathjs\lib\cjs\expression\parse.js:1035:12)
```

**Comment by m93a** [comment]
Good catch! This is definitely a bug in the parser.

I found another unintuitive thing: expressions like `a.2` are parsed as implicit multiplictation `a * 0.2`. Maybe the two effects are related. After all, you can do this:
```
b = 2
   → 2
b.1e2
   → 20
```
So maybe the parser mistakes “`.e`” for the exponential notation and complains that there must be a digit between “`.`” and “`e`”.

**Comment by josdejong** [comment]
Fixed via c8742d1 (not yet published)

**Comment by josdejong** [comment]
Fixed now in `v9.4.3`
